lib/internal/process/execution.js

```diff
@@ -33,6 +33,24 @@ function tryGetCwd() {
   }
 }
 
+function evalModule(source) {
+  const { decorateErrorStack } = require('internal/util');
+  const asyncESM = require('internal/process/esm_loader');
+  asyncESM.loaderPromise.then(async (loader) => {
+    const { result } = await loader.eval(source);
+    if (require('internal/options').getOptionValue('--print')) {
+      console.log(result);
+    }
+  })
+  .catch((e) => {
+    decorateErrorStack(e);
+    console.error(e);
+    process.exit(1);
+  });
+  // Handle any nextTicks added in the first tick of the program.
+  process._tickCallback();
```

Comment:
this feels a little fragile, do we have a unit test around this specific behavior (I think it would be worthwhile because someone will eventually try removing this function).

Reply:
TBH I'm not sure off hand exactly how to test this behavior, any suggestions?

/cc @guybedford

Reply:
yeah this is tested, we've had it for a really really long time, it's not added by this pr.